src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -480,35 +183,66 @@ public async Task InitializeFromProfile(DacDeployOptions options, string profile
+        /// </summary>
+        /// <param name="options"></param>
+        public void InitializeBooleanTypeOptions(DacDeployOptions options)
+        {
+            // To fill the options map table directly from the boolean type DacDeployoptions
+            PropertyInfo[] dacDeploymentOptionsProperties = options.GetType().GetProperties();
+            foreach (PropertyInfo  prop in dacDeploymentOptionsProperties)
+            {
+                if (prop.PropertyType == typeof(System.Boolean))
+                {
+                    object setProp = GetDeploymentOptionProp(prop, options);
+                    this.BooleanOptionsDictionary[prop.Name] = (DeploymentOptionProperty<bool>)setProp;
+                }
+            }
+        }
+
         public void SetOptions(DacDeployOptions options)
         {
             System.Reflection.PropertyInfo[] deploymentOptionsProperties = this.GetType().GetProperties();
+            // Set the default options properties
+            InitializeBooleanTypeOptions(options);
+            InitializeNonBooleanTypeOptions(options);
+        }
 
-            foreach (var deployOptionsProp in deploymentOptionsProperties)
+        /// <summary>
+        /// Preparing all non boolean properties (except BooleanOptionsDictionary)
+        /// </summary>
+        /// <param name="options"></param>
+        public void InitializeNonBooleanTypeOptions(DacDeployOptions options)
+        {
+            // preparing remaining properties
+            PropertyInfo[] deploymentOptionsProperties = this.GetType().GetProperties();
+            foreach (PropertyInfo deployOptionsProp in deploymentOptionsProperties)
             {
-                var prop = options.GetType().GetProperty(deployOptionsProp.Name);
-                // Note that we set excluded object types here since dacfx has this value as null;
-                if (prop != null)
+                if (deployOptionsProp.Name != nameof(DeploymentOptions.BooleanOptionsDictionary))
                 {
-                    SetGenericDeployOptionProps(prop, options, deployOptionsProp);
+                    var prop = options.GetType().GetProperty(deployOptionsProp.Name);
+                    object setProp = GetDeploymentOptionProp(prop, options);
+                    deployOptionsProp.SetValue(this, setProp);
                 }
             }
         }
 
         /// <summary>
-        /// Sets the Value and Description to all properties
+        /// Prepares and returns the value and description of a property
         /// </summary>
         /// <param name="prop"></param>
-        /// <param name="val"></param>
-        /// <param name="deployOptionsProp"></param>
-        public void SetGenericDeployOptionProps(PropertyInfo prop, DacDeployOptions options, PropertyInfo deployOptionsProp)
+        /// <param name="options"></param>
+        public object GetDeploymentOptionProp(PropertyInfo? prop, DacDeployOptions options)
```

Comment:
what's the `?` after `PropertyInfo?` for? Could `prop` be null?

Reply:
Not expecting null here! but i see intellisense green line warning "'prop' may be null here"! Should i consider them?

Reply:
If intellisense is telling you that it can be null that means something upstream could potentially be returning null. 

In this case it's because GetProperty can return null if that property doesn't exist on the type - so you should make sure you're handling that appropriately.